src/main/java/frc/robot/subsystems/pivot/Pivot.java

```diff
@@ -0,0 +1,161 @@
+    // instances, we may need to have a goal that dynamically changes based on a state from another
+    // class. If we do not use a Supplier, java will only create an instance of the setpoint once
+    // and never check if it is updated again.
+    Supplier<Rotation2d> goal;
+
+    PivotGoal(Supplier<Rotation2d> goal) {
+      this.goal = goal;
+    }
+
+    public Rotation2d getGoal() {
+      return this.goal.get();
+    }
+  }
+
+  // Hardware is where we interact with the motors and sensors directly
+  private final PivotKrakenHardware kPivotHardware;
+  // "Inputs" is telemetry data from the motors or sensors, such as motor temperature
+  private final PivotInputsAutoLogged kPivotInputs = new PivotInputsAutoLogged();
+
+  @AutoLogOutput(key = "Pivot/CurrentGoal")
+  private PivotGoal currentPivotGoal = null;
+
+  @AutoLogOutput(key = "Pivot/CurrentGoalPosition")
+  private Rotation2d currentPivotGoalPosition = new Rotation2d();
+
+  private final LoggedTunableNumber kP =
+      new LoggedTunableNumber("Pivot/Gains/kP", PivotConstants.kPivotGains.kP());
+  private final LoggedTunableNumber kI =
+      new LoggedTunableNumber("Pivot/Gains/kI", PivotConstants.kPivotGains.kI());
+  private final LoggedTunableNumber kD =
+      new LoggedTunableNumber("Pivot/Gains/kD", PivotConstants.kPivotGains.kD());
+  private final LoggedTunableNumber kS =
+      new LoggedTunableNumber("Pivot/Gains/kS", PivotConstants.kPivotGains.kS());
+  private final LoggedTunableNumber kV =
+      new LoggedTunableNumber("Pivot/Gains/kV", PivotConstants.kPivotGains.kV());
+  private final LoggedTunableNumber kA =
+      new LoggedTunableNumber("Pivot/Gains/kA", PivotConstants.kPivotGains.kA());
+  private final LoggedTunableNumber kG =
+      new LoggedTunableNumber("Pivot/Gains/kG", PivotConstants.kPivotGains.kG());
+  private final LoggedTunableNumber kMaxVelocity =
+      new LoggedTunableNumber(
+          "Pivot/MotionMagic/kMaxVelocity", PivotConstants.kPivotGains.kMaxVelocity());
+  private final LoggedTunableNumber kMaxAcceleration =
+      new LoggedTunableNumber(
+          "Pivot/MotionMagic/kMaxAcceleration", PivotConstants.kPivotGains.kMaxAcceleration());
+
+  /** Creates a new Pivot. */
+  public Pivot(PivotKrakenHardware pivotHardware) {
+    kPivotHardware = pivotHardware;
+  }
+
+  @Override
+  public void periodic() {
+    // We pass in kPivotInputs to kPivotHardware to "update" it with data from the hardware layer
+    // (E.g. Motor temperatures, motor positions). This must be called periodically in order to
+    // properly update and call values from kPivotInputs.
+    kPivotHardware.updateInputs(kPivotInputs);
+    Logger.processInputs("Pivot/Inputs", kPivotInputs);
+
+    if (currentPivotGoal != null) {
+      currentPivotGoalPosition = currentPivotGoal.getGoal();
+      setPosition(currentPivotGoalPosition.getRadians());
+    }
+
+    // Update feedback, feedforward, and motion magic gains if we change them from network tables
```

Comment:
This is another place where an ellipsed method signature is used, so the role of this function isn't immediately evident to someone new to the code.  A comment to guide readers would help.